src/sqs/discovery.ts

```diff
@@ -1,12 +1,18 @@
-import {MessageHandler} from "./index"
+import { MessageHandler } from "./index";
 import app from "../worker/app";
-import {discovery} from "../sync/discovery";
+import { discovery, discoveryOctoKit } from "../sync/discovery";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
 
 export type DiscoveryMessagePayload = {
 	installationId: number,
 	jiraHost: string
 }
 
 export const discoveryQueueMessageHandler : MessageHandler<DiscoveryMessagePayload> = async (context) => {
-	await discovery(app)({data: context.payload}, context.log);
-}
+	const useNewGHClient = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_DISCOVERY, false, context?.payload?.jiraHost);
```

Comment:
Moved the feature flag up a level so that the original discovery(now discoveryOctokit) can stay as it was and the new GitHub client version can be a separate function entirely.